src/TrapEndcapTracker_geo.cpp

```diff
@@ -58,24 +50,19 @@ static Ref_t create_detector(Detector& description, xml_h e, SensitiveDetector s
   map<string, std::array<double, 2>> module_thicknesses;
   PlacedVolume                       pv;
 
-  // ACTS extension
-  {
-    Acts::ActsExtension* detWorldExt = new Acts::ActsExtension();
-    detWorldExt->addType("endcap", "detector");
-    // SJJ probably need to set the envelope here, as ACTS can't figure
-    // that out for Assembly volumes. May also need binning to properly pick up
-    // on the support material @TODO
-    //
-    // Add the volume boundary material if configured
-    for (xml_coll_t bmat(x_det, _Unicode(boundary_material)); bmat; ++bmat) {
-      xml_comp_t x_boundary_material = bmat;
-      Acts::xmlToProtoSurfaceMaterial(x_boundary_material, *detWorldExt, "boundary_material");
-    }
-    sdet.addExtension<Acts::ActsExtension>(detWorldExt);
+  // Set detector type flag
+  dd4hep::xml::setDetectorTypeFlag(x_det, sdet);
+  auto &params = DD4hepDetectorHelper::ensureExtension<dd4hep::rec::VariantParameters>(
+      sdet);
+
+  // Add the volume boundary material if configured
+  for (xml_coll_t bmat(x_det, _Unicode(boundary_material)); bmat; ++bmat) {
+    xml_comp_t x_boundary_material = bmat;
+    DD4hepDetectorHelper::xmlToProtoSurfaceMaterial(x_boundary_material, params,
+                                         "boundary_material");
   }
 
   assembly.setVisAttributes(description.invisible());
-  sens.setType("tracker");
```

Comment:
Are we sure we should remove this? This is unrelated to Acts.

Reply:
Moved it to the top for clarity. Only needs to be invoked once.